x-pack/plugins/infra/public/pages/logs/log_entry_categories/sections/notices/quality_warning_notices.tsx

```diff
@@ -0,0 +1,98 @@
+ */
+
+import { EuiCallOut } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import React from 'react';
+import { CategoryQualityWarningReason, QualityWarning } from './quality_warnings';
+
+export const CategoryQualityWarnings: React.FC<{ qualityWarnings: QualityWarning[] }> = ({
+  qualityWarnings,
+}) => (
+  <>
+    {qualityWarnings.map((qualityWarning, qualityWarningIndex) => (
+      <EuiCallOut
+        key={`${qualityWarningIndex}`}
+        title={categoryQualityWarningCalloutTitle}
+        color="warning"
+        iconType="alert"
+      >
+        <p>
+          <FormattedMessage
+            id="xpack.infra.logs.logEntryCategories.categoryQualityWarningCalloutMessage"
+            defaultMessage="While analyzing the log messages we've detected some problems which might indicate a reduced quality of the categorization results."
+          />
+        </p>
+        <ul>
+          {qualityWarning.reasons.map((reason, reasonIndex) => (
+            <li key={`${reasonIndex}`}>
+              <CategoryQualityWarningReasonDescription reason={reason} />
+            </li>
+          ))}
+        </ul>
+      </EuiCallOut>
+    ))}
+  </>
+);
+
+const categoryQualityWarningCalloutTitle = i18n.translate(
+  'xpack.infra.logs.logEntryCategories.categoryQUalityWarningCalloutTitle',
+  {
+    defaultMessage: 'Quality warning',
+  }
+);
+
+const CategoryQualityWarningReasonDescription: React.FC<{
+  reason: CategoryQualityWarningReason;
+}> = ({ reason }) => {
+  switch (reason.type) {
+    case 'singleCategory':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.logEntryCategories.singleCategoryWarningReasonDescription"
+          defaultMessage="The analysis couldn't extract more than a single category from the log message."
+        />
+      );
+    case 'manyRareCategories':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.logEntryCategories.manyRareCategoriesWarningReasonDescription"
+          defaultMessage="{rareCategoriesRatio, number, percent} of the categories only rarely have messages assigned to them."
+          values={{
+            rareCategoriesRatio: reason.rareCategoriesRatio,
+          }}
+        />
+      );
+    case 'manyCategories':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.logEntryCategories.manyCategoriesWarningReasonDescription"
+          defaultMessage="The ratio of categories per analyzed document is very high with {categoriesDocumentRatio, number }."
+          values={{
+            categoriesDocumentRatio: reason.categoriesDocumentRatio.toFixed(2),
+          }}
+        />
+      );
+    case 'noFrequentCategories':
+      return (
+        <FormattedMessage
+          id="xpack.infra.logs.logEntryCategories.noFrequentCategoryWarningReasonDescription"
+          defaultMessage="None of the extracted categories frequently have message assigned to them."
```

Comment:
```suggestion
          defaultMessage="None of the extracted categories frequently have messages assigned to them."
```
??? or do you mean the message field with that?

Reply:
No, that was a typo - thanks for catching it